crates/evm-storage-proofs/src/client.rs

```diff
@@ -0,0 +1,119 @@
+use alloy::{providers::Provider, rpc::types::EIP1186AccountProofResponse};
+use alloy_primitives::{Address, FixedBytes, hex::FromHex};
+use anyhow::{Context, Result};
+
+pub type DefaultProvider = alloy::providers::fillers::FillProvider<
```

Comment:
I remember stepping through this chain of types before, its kinda mind blowing 🤯 

good call on a type alias lol